editor/plugins/org.fusesource.ide.camel.editor/src/org/fusesource/ide/camel/editor/preferences/PreferredLabelEditor.java

```diff
@@ -0,0 +1,424 @@
+				} else if (widget == editButton) {
+					editPressed();
+				} else if (widget == removeButton) {
+					removePressed();
+				} else if (widget == table) {
+					selectionChanged();
+				}
+			}
+		};
+	}
+
+	protected void addPressed() {
+		PreferredLabelDialog dialog = new PreferredLabelDialog(getShell());
+		dialog.setComponentValidator(new ComponentValidator(getComponents()));
+		dialog.setParameterValidator(new ParameterValidator());
+		if (dialog.open() == Window.OK) {
+			addRow(dialog.getComponent(), dialog.getParameter());
+		}
+	}
+
+	protected void editPressed() {
+		int index = table.getSelectionIndex();
+		if (index >= 0) {
+			TableItem selectedItem = table.getItem(index);
+			PreferredLabelDialog dialog = new PreferredLabelDialog(getShell());
+			dialog.setComponentValidator(new ComponentValidator());
+			dialog.setParameterValidator(new ParameterValidator());
+			dialog.setComponent(selectedItem.getText(0));
+			dialog.setParameter(selectedItem.getText(1));
+			if (dialog.open() == Window.OK) {
+				updateRow(index, dialog.getComponent(), dialog.getParameter());
+			}
+		}
+	}
+
+	protected void removePressed() {
+		table.remove(table.getSelectionIndices());
+	}
+
+	@Override
+	protected void doLoad() {
+		if (table != null) {
+			String s = getPreferenceStore().getString(getPreferenceName());
+			parseString(s);
+		}
+	}
+
+	@Override
+	protected void doLoadDefault() {
+		if (table != null) {
+			table.removeAll();
+			String s = getPreferenceStore().getDefaultString(getPreferenceName());
+			parseString(s);
+		}
+	}
+
+	@Override
+	protected void doStore() {
+		String s = createList(table.getItems());
+		if (s != null) {
+			getPreferenceStore().setValue(getPreferenceName(), s);
+		}
+	}
+
+	/**
+	 * Splits the given string into a list of strings. This method is the converse of <code>createList</code>.
+	 * <p>
+	 * Subclasses must implement this method.
+	 * </p>
+	 *
+	 * @param stringList
+	 *            the string
+	 * @return an array of <code>String</code>
+	 * @see #createList
+	 */
+	private void parseString(String stringList) {
+		if (stringList.isEmpty()) {
+			return;
+		}
+		String[] preferredLabels = stringList.split(AbstractCamelModelElement.USER_LABEL_DELIMETER);
+		Stream.of(preferredLabels).forEach(label -> addRow(label.split("\\.")));
+	}
+
+	/**
+	 * Combines the given list of items into a single string. This method is the converse of <code>parseString</code>.
+	 * <p>
+	 * Subclasses must implement this method.
+	 * </p>
+	 *
+	 * @param items
+	 *            the list of items
+	 * @return the combined string
+	 * @see #parseString
+	 */
+	private String createList(TableItem[] rows) {
+		return String.join(AbstractCamelModelElement.USER_LABEL_DELIMETER, getPreferredLabels());
+	}
+
+	public List<String> getPreferredLabels() {
+		List<String> preferredLabels = new ArrayList<>();
+		Stream.of(table.getItems()).map(row -> row.getText(0) + "." + row.getText(1)).forEach(preferredLabels::add);
```

Comment:
I think that you don't need to create a new List and replace .forEach(preferredLabels::add) by .collect(Collectors.toList())